spec/models/application_spec.rb

```diff
@@ -0,0 +1,29 @@
+require "rails_helper"
+
+RSpec.describe Application, type: :model do
+  describe "relationships" do
+    it { should have_many(:application_pets) }
+    it { should have_many(:pets).through(:application_pets) }
+  end
+
+  describe "validations" do
+    it { should validate_presence_of(:name) }
+    it { should validate_presence_of(:street_address) }
+    it { should validate_presence_of(:city) }
+    it { should validate_presence_of(:state) }
+    it { should validate_presence_of(:zip_code) }
+    it { should validate_presence_of(:description) }
+    it { should validate_presence_of(:status) }
+  end
+
+  describe ".search_pets_by_name(pet_name)" do 
+    it "should return pets that have been searched" do 
```

Comment:
Nice work testing this model method! You can  use this same test and just change your implementation to use the already made .search